lib/matplotlib/figure.py

```diff
@@ -2614,12 +2614,16 @@ def show(self, warn=True):
             may only be shown briefly or not shown at all if you or your
             environment are not managing an event loop.
 
-            Proper use cases for `.Figure.show` include running this from a
-            GUI application or an IPython shell.
-
-            If you're running a pure python shell or executing a non-GUI
-            python script, you should use `matplotlib.pyplot.show` instead,
-            which takes care of managing the event loop for you.
+            Proper use cases for `.Figure.show` include running this from a GUI
+            application (where there is persistently an event loop running) or
+            from a shell with in input hook configured (IPython will do this).
+            Some, but not all, GUI toolkits will register an input hook on
+            import.  See :ref:`cp_integration` for more details.
```

Comment:
My apologies, but I find these changes much more confusing than the original.  What is an "input hook"?  What is a "GUI toolkit", and why do they register an "input hook"?  Where do they register it? and what does "on import" mean?  On import of what?

Reply:
Note that there is a typo in the previous paragraph "an GUI even loop"...

Reply:
It does include a link to the (much) longer explanation. 

#22169 noted that they found a case where it _did_ work.  The goal was to add a few more details about when you should expect this to work or not.